server/etcdserver/server.go

```diff
@@ -1081,9 +1084,10 @@ func (s *EtcdServer) applyEntries(ep *etcdProgress, apply *apply) {
 }
 
 func (s *EtcdServer) triggerSnapshot(ep *etcdProgress) {
-	if ep.appliedi-ep.snapi <= s.Cfg.SnapshotCount {
+	if !s.forceSnapshot && ep.appliedi-ep.snapi <= s.Cfg.SnapshotCount {
```

Comment:
There is potential corner case when:
  s.forceSnapshot && (ep.appliedi=ep.snapi)

Not sure if we can trigger this in practice, but more defensive would be NOT to trigger snapshot in such case
(i.e. overwrite existing snapshot).

Reply:
Done